Framework/Core/include/Framework/LocalRootFileService.h

```diff
@@ -0,0 +1,33 @@
+// Copyright CERN and copyright holders of ALICE O2. This software is
+// distributed under the terms of the GNU General Public License v3 (GPL
+// Version 3), copied verbatim in the file "COPYING".
+//
+// See https://alice-o2.web.cern.ch/ for full licensing information.
+//
+// In applying this license CERN does not waive the privileges and immunities
+// granted to it by virtue of its status as an Intergovernmental Organization
+// or submit itself to any jurisdiction.
+#ifndef FRAMEWORK_LOCALROOTFILESERVICE_H
+#define FRAMEWORK_LOCALROOTFILESERVICE_H
+
+#include "Framework/RootFileService.h"
+#include "TFile.h"
+
+#include <map>
+#include <string>
+#include <vector>
+#include <memory>
+
+namespace o2 {
+namespace framework {
+
+/// A simple service to create ROOT files in the local folder
+class LocalRootFileService : public RootFileService {
+public:
+  std::shared_ptr<TFile> open(const char *fmt, ...) final;
+  char *format(const char *fmt, ...) final;
```

Comment:
here I get the following warning
```
In file included from /home/richter/src/alisw/sw/SOURCES/O2/dev/0/Framework/Core/src/runDataProcessing.cxx:24:0:
/home/richter/src/alisw/sw/SOURCES/O2/dev/0/Framework/Core/include/Framework/LocalRootFileService.h:28:9: warning: invalid covariant return type for ‘virtual char* o2::framework::LocalRootFileService::format(const char*, ...)’
   char *format(const char *fmt, ...) final;
         ^~~~~~
In file included from /home/richter/src/alisw/sw/SOURCES/O2/dev/0/Framework/Core/include/Framework/LocalRootFileService.h:13:0,
                 from /home/richter/src/alisw/sw/SOURCES/O2/dev/0/Framework/Core/src/runDataProcessing.cxx:24:
/home/richter/src/alisw/sw/SOURCES/O2/dev/0/Framework/Core/include/Framework/RootFileService.h:32:23: note:   overriding ‘virtual const char* o2::framework::RootFileService::format(const char*, ...)’
   virtual const char *format(const char *fmt, ...) = 0;
                       ^~~~~~
```

Reply:
The base class `RootFileService` defines the return type of `format` as `const char*`. That is where the warning comes from.